components/datetime/src/skeleton/helpers.rs

```diff
@@ -485,7 +484,7 @@ pub fn get_best_available_format_pattern<'data>(
         }
     }
 
-    #[allow(clippy::expect_used)] // TODO(#1668) Clippy exceptions need docs or fixing.
+    #[allow(clippy::expect_used)] // We always have at least one format pattern.
```

Comment:
Issue: How are you sure about this? For example, is it possible that if the pattern is missing from data, no pattern would be found? If the lookup is guaranteed to be successful, can you express this somehow in code (like having a code block return the value), rather than using an `Option` in the first place?

Reply:
Was able to refactor it away.